### PR TITLE
Use anyhow's ensure and Context where appropriate.

### PR DESCRIPTION
**Pull request by afck**
## Motivation

Compared to using `anyhow!` or `bail!` directly,
* The `anyhow::Context` methods are more concise and preserve more information about the original error. (The resulting `anyhow::Error` can still be downcast.)
* `anyhow::ensure!` is more concise and—at least if the condition is not a negation—often more readable.

## Proposal

Replace some usages of `anyhow!` and `bail!`.

Import `Context as _` to avoid confusion with `linera-views`' `Context` trait.

Qualify `anyhow::ensure!` to avoid confusion with `linera-base`'s `ensure!` macro.
We could extend the latter at some point to accept formatting arguments.

## Test Plan

No logic has changed.

## Links

- Related discussion: https://github.com/linera-io/linera-protocol/pull/1107#discussion_r1353232197
- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: linera-explorer/src/lib.rs [hunk 1]
```diff
@@ -567,13 +567,11 @@ async fn page(
...
-        "application" => match find_arg(args, "app").map(|v| parse(&v)) {
-            None => Err(anyhow!("unknown application")),
-            Some(app_js) => {
-                let app = from_value::<Application>(app_js).unwrap();
-                application(app).await
-            }
-        },
+        "application" => {
+            let app_arg = find_arg(args, "app").context("unknown application")?;
+            let app = from_value::<Application>(parse(&app_arg)).unwrap();
```
Comment (christos-h) [inline linera-explorer/src/lib.rs:572]: Why the `unwrap` here?

File: linera-service/src/config.rs [hunk 2]
```diff
@@ -229,9 +227,12 @@ impl WalletState {
     }
 
     pub fn set_default_chain(&mut self, chain_id: ChainId) -> Result<(), anyhow::Error> {
-        if !self.inner.chains.contains_key(&chain_id) {
-            bail!("Chain {} cannot be assigned as the default chain since it does not exist in the wallet.", &chain_id);
-        }
+        anyhow::ensure!(
```
Comment (christos-h) [inline linera-service/src/config.rs:230]: Nice, didn't know about this one.

File: linera-service/src/storage.rs [hunk 3]
```diff
@@ -128,30 +129,19 @@ impl FromStr for StorageConfig {
...
-                            let Some(port_str) = parts.next() else {
-                                bail!(err_msg);
-                            };
-                            let Ok(_num_port) = port_str.parse::<u16>() else {
-                                bail!(err_msg);
-                            };
-                            if uri.is_some() {
-                                bail!("The uri has already been assigned");
-                            }
+                            let empty = parts.next().context(err_msg)?;
```
Comment (christos-h) [inline linera-service/src/storage.rs:132]: Heh nice, already did it in an upcoming PR for Scylla.

**Review by christos-h — APPROVED** [review]
Just questioning the unwrap. Which I guess is the old behavior so approved :)

File: linera-explorer/src/lib.rs (hunk 1 above)
Comment (afck) [reply linera-explorer/src/lib.rs:572]: I didn't add that.
But you're right, we could probably return an error instead.

Comment (afck) [reply linera-explorer/src/lib.rs:572]: Ironically, I would actually have to use `map_err(|error| anyhow!(…))?` here!
But all the other places in this file where `from_value` is called use `expect`, so I'd rather change the `unwrap` to an `expect` for now, for consistency.

**Review by ma2bd — APPROVED** [review]

**Pull request merged**